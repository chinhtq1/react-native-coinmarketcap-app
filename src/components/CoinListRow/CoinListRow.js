import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import FastImage from 'react-native-fast-image'

export const COIN_ROW_HEIGHT = 64

// FastImage added just for example here, cryptoicons.org sometimes not working
const Row = ({ index, name, price, symbol = '', volume }) => (
  <View style={styles.container}>
    <Text style={styles.index}>{index}</Text>
    <FastImage
      style={styles.dummyImage}
      source={{ uri: `https://cryptoicons.org/api/icon/${symbol.toLowerCase()}/200` }}
      resizeMode={FastImage.resizeMode.contain}
    />
    <View style={styles.dataContainer}>
      <View style={styles.spaceBetween}>
        <Text style={styles.topText}>{name}</Text>
        <Text style={styles.topText}>{price}</Text>
      </View>
      <View style={styles.spaceBetween}>
        <Text style={styles.bottomText}>{symbol}</Text>
        <Text style={styles.bottomText}>Vol {volume}</Text>
      </View>
    </View>
  </View>
)

const styles = StyleSheet.create({
  container: {
    height: COIN_ROW_HEIGHT,
    flexDirection: 'row',
    paddingHorizontal: 16,
    alignItems: 'center',
  },
  dataContainer: {
    justifyContent: 'center',
    flex: 1,
    height: 44,
  },
  spaceBetween: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-between',
    flexDirection: 'row',
  },
  nameContainer: {
    flex: 1,
  },
  bottomText: {
    color: '#747474',
  },
  index: {
    width: 24,
    color: '#858585',
    fontSize: 12,
  },
  topText: {
    fontSize: 16,
    fontWeight: '500',
  },
  dummyImage: {
    width: 32,
    height: 32,
    borderRadius: 16,
    marginRight: 12,
    backgroundColor: '#efefef',
  },
})

export const CoinListRow = React.memo(Row)
