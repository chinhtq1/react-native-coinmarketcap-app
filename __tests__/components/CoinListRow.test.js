import React from 'react'
import renderer from 'react-test-renderer'
import { CoinListRow } from '../../src/components/CoinListRow'

const DATA = {
  index: 22,
  symbol: 'BTC',
  name: 'Bitcoin',
  price: '333.22',
  volume: '333k',
}

test('renders correctly empty component', () => {
  const component = <CoinListRow />
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders with data', () => {
  const component = <CoinListRow {...DATA} />
  const tree = renderer.create(component).toJSON()
  expect(tree).toMatchSnapshot()
})
