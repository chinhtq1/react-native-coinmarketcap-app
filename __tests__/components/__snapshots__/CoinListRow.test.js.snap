// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly empty component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 64,
      "paddingHorizontal": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#858585",
        "fontSize": 12,
        "width": 24,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#efefef",
          "borderRadius": 16,
          "height": 32,
          "marginRight": 12,
          "width": 32,
        },
      ]
    }
  >
    <FastImageView
      resizeMode="contain"
      source={
        Object {
          "uri": "https://cryptoicons.org/api/icon//200",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "height": 44,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#747474",
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#747474",
          }
        }
      >
        Vol 
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders with data 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 64,
      "paddingHorizontal": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#858585",
        "fontSize": 12,
        "width": 24,
      }
    }
  >
    22
  </Text>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#efefef",
          "borderRadius": 16,
          "height": 32,
          "marginRight": 12,
          "width": 32,
        },
      ]
    }
  >
    <FastImageView
      resizeMode="contain"
      source={
        Object {
          "uri": "https://cryptoicons.org/api/icon/btc/200",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "height": 44,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Bitcoin
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        333.22
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#747474",
          }
        }
      >
        BTC
      </Text>
      <Text
        style={
          Object {
            "color": "#747474",
          }
        }
      >
        Vol 
        333k
      </Text>
    </View>
  </View>
</View>
`;
